File: music-api/src/main/java/com/music/sale/web/user/request/UpdateUserProfileRequest.java
package com.music.sale.web.user.request;

import com.music.sale.application.user.dto.UpdateUserProfileInput;
import com.music.sale.domain.user.enums.Gender;
import java.time.LocalDate;

public record UpdateUserProfileRequest(
        String name,
        String phoneNumber,
        Gender gender,
        String birthDate,
        String profileImage,
        String bio) {

    public UpdateUserProfileInput toInput() {
        LocalDate parsedBirthDate = birthDate != null ? LocalDate.parse(birthDate) : null;
        return new UpdateUserProfileInput(
                name,
                phoneNumber,
                gender,
                parsedBirthDate,
                profileImage,
                bio);
    }
}

File: music-api/src/main/kotlin/com/music/sale/web/user/request/UpdateUserProfileRequest.kt
package com.music.sale.web.user.request

import com.music.sale.application.user.dto.UpdateUserProfileInput
import com.music.sale.domain.user.enum.Gender

data class UpdateUserProfileRequest(
    val name: String? = null,
    val phoneNumber: String? = null,
    val gender: Gender? = null,
    val birthDate: String? = null,
    val profileImage: String? = null,
    val bio: String? = null,
) {
    fun toInput(): UpdateUserProfileInput {
        return UpdateUserProfileInput(
            name = name,
            phoneNumber = phoneNumber,
            gender = gender,
            birthDate = birthDate?.let { java.time.LocalDate.parse(it) },
            profileImageUrl = profileImage,
            bio = bio,
        )
    }
}
